cardiopy/ekg.py

```diff
@@ -196,7 +196,13 @@ def detect_Rpeaks(self, smooth):
         
         peaks = []
         x = 0
-        while x < len(raw):
+        if raw[len(raw)-1] > thres[len(raw)-1]:
+            for h in range(len(raw)-1, 0, -1):
+                if raw[len(raw)-(h+1)] < thres[len(raw)-(h+1)]:
+                    end = len(raw) - (h+1)
+        else:
+           end = len(raw)
```

Comment:
```suggestion
        # if final value is greater than threshold
        if raw[len(raw)-1] > thres[len(raw)-1]:
            # walk backwards from final value to find previous threshold crossing
            for h in range(len(raw)-1, 0, -1):
                if raw[len(raw)-(h+1)] < thres[len(raw)-(h+1)]:
                    # set end pointer to point prior to last positive threshold crossing
                    end = len(raw) - (h+1)
        # otherwise set end point to end of data
        else:
           end = len(raw)
```